airflow/www/security.py

```diff
@@ -728,3 +735,102 @@ def check_authorization(
                 return False
 
         return True
+
+    # TODO: Whether to create APISecurityManager and move api related code to it?
+    def is_user_logged_in(self):
+        """Raise if user already logged in"""
+        if g.user is not None and g.user.is_authenticated:
+            raise Unauthenticated(detail="Client already authenticated")  # For security
```

Comment:
This should be a "400 Bad Request".

Reply:
I'm actually returning 401 for all failure due to your comment here: https://github.com/apache/airflow/pull/15042#discussion_r603960819